scripts/world_bank/worldbank.py

```diff
@@ -47,222 +47,347 @@
+            indicator_code: Dataframe with INDICATOR_CODES to include.
+
+        Returns:
+            worldbank_dataframe: A tidied pandas dataframe where each row has
+            the format (indicator code, country ISO, year, value)
+            for all countries and all indicators provided.
+    """
+    worldbank_dataframe = pd.DataFrame()
+    indicators_to_keep = list(indicator_codes['IndicatorCode'].unique())
+
+    for index, country_code in enumerate(worldbank_countries['CountryCode']):
+        print(f"Downloading {country_code}")
+        country_df = read_worldbank(country_code)
+
+        # Remove unneccessary indicators.
+        country_df = country_df[country_df['IndicatorCode']
+                                .isin(indicators_to_keep)]
+
+        # Map country codes to ISO.
+        country_df['IsoCode'] = country_code
+
+        # Add new row to main datframe.
+        worldbank_dataframe = worldbank_dataframe.append(country_df)
+
+    # Map indicator codes to unique Statistical Variable.
+    worldbank_dataframe['StatisticalVariable'] = (
+        worldbank_dataframe['IndicatorCode'].apply(
+            lambda code: f"WorldBank/{code.replace('.', '_')}"))
+    worldbank_dataframe = worldbank_dataframe.rename({'year': 'Year'}, axis=1)
+    return worldbank_dataframe
+
+
+def output_csv_and_tmcf_by_grouping(worldbank_dataframe, tmcfs_for_stat_vars, indicator_codes):
+    """ Outputs TMCFs and CSVs for each grouping of stat vars.
+
+        Args:
+            worldbank_dataframe -> Dataframe containing all indicators for all
+            countries tmcfs_for_stat_vars -> Array of tuples of templace MCF,
+                columns on stat var observations,
+                indicator codes for that template.
+            MCF indicator_codes -> Dataframe with INDICATOR_CODES to include.
+    """
+    # Only include a subset of columns in the final csv
+    output_csv = worldbank_dataframe[['StatisticalVariable', 'IndicatorCode', 'IsoCode', 'Year', 'Value']]
+
+    # Output tmcf and csv for each unique WorldBank grouping.
+    for index, enum in enumerate(tmcfs_for_stat_vars):
+        tmcf, stat_var_obs_cols, stat_vars_in_group = enum
+        if len(stat_vars_in_group) != 0:
+            with open(f"output/WorldBank_{index}.tmcf", 'w', newline='') as f_out:
+                f_out.write(tmcf)
+                f_out.write(COUNTRY_MCF)
+
+            # Output only the indicator codes in that grouping.
+            matching_csv = output_csv[
+                output_csv['IndicatorCode'].isin(stat_vars_in_group)]
+
+            # Include the Stat Observation columns in the output CSV.
+            if len(stat_var_obs_cols) > 1:
+                matching_csv = pd.merge(matching_csv,
+                                        indicator_codes[stat_var_obs_cols],
+                                        on="IndicatorCode")
+            
+            # Format to decimals.
+            matching_csv = matching_csv.round(10)
```

Comment:
Why is rounding necessary?  Is there some data flaw that is being fixed?

If it is just to make them pretty, it would be better to leave the raw values.  The values can be prettied when an application decides to show values to a human.